### PR TITLE
Add server-side aspects of Step Vnext

### PR DESCRIPTION
**Pull request by rain-on**
This includes only the BuildInformation sub-step, along with connection oriented data.

This change creates the JSPs used to enter user data for the BuildInformation process, validate the fields, and ensure they are correctly rendered in:
* Edit page
* View page
*  Parameter description

### CODE REVIEW

File: octopus-agent/src/main/resources/META-INF/build-agent-plugin-Octopus.TeamCity.xml [hunk 1]
```diff
@@ -28,4 +28,5 @@
     <bean class="octopus.teamcity.agent.OctopusPromoteReleaseRunner"/>
     <bean class="octopus.teamcity.agent.OctopusPushPackageRunner"/>
     <bean class="octopus.teamcity.agent.OctopusPackPackageRunner"/>
+    <bean class="octopus.teamcity.agent.generic.OctopusGenericRunner"/>
```
Comment (long-sean) [inline octopus-agent/src/main/resources/META-INF/build-agent-plugin-Octopus.TeamCity.xml:31]: Would not registering the generic runner hide vNext build steps as per OctopusGenericRunner message "Octopus Generic Runner is not yet ready for use, please use existing steps"?

File: octopus-server/src/main/resources/buildServerResources/v2/subpages/editBuildInformationParameters.jsp [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+<l:settingsGroup title="Package">
+
+    <tr>
+        <th>Package IDs:<l:star/></th>a
+        <td>
+            <props:multilineProperty name="${keys.packageIdKey}" rows="5" cols="50" linkTitle="Package IDs"
+                                     expanded="true"/>
+            <span class="error" id="error_${keys.packageIdKey}"></span>
+            <span class="smallNote">
```
Comment (long-sean) [inline octopus-server/src/main/resources/buildServerResources/v2/subpages/editBuildInformationParameters.jsp:36]: Weird formatting?

File: octopus-agent/src/main/resources/META-INF/build-agent-plugin-Octopus.TeamCity.xml (hunk 1 above)
Comment (rain-on) [reply octopus-agent/src/main/resources/META-INF/build-agent-plugin-Octopus.TeamCity.xml:31]: So - the build-step will always fail validation (as we raise an invalidProperty when validating  - so technically, we probably don't need to have any agent stuff at all if we didn't want to

File: octopus-server/src/main/resources/buildServerResources/v2/subpages/editBuildInformationParameters.jsp (hunk 2 above)
Comment (rain-on) [reply octopus-server/src/main/resources/buildServerResources/v2/subpages/editBuildInformationParameters.jsp:36]: cleaned up - but not sure what it SHOULD look like.

**Review by long-sean — APPROVED** [review]
good to go

File: octopus-common/src/main/java/octopus/teamcity/common/commonstep/CommonStepPropertyKeys.java [hunk 3]
```diff
@@ -36,19 +36,19 @@ public String getStepTypeKey() {
     return STEP_TYPE;
   }
 
-  public String getServerKey() {
+  public String getServerUrlKey() {
     return SERVER_URL;
   }
 
-  public String getApiKey() {
+  public String getApiKeyKey() {
```
Comment (long-sean) [inline octopus-common/src/main/java/octopus/teamcity/common/commonstep/CommonStepPropertyKeys.java:43]: KeyKey

**Review by long-sean — APPROVED** [review]

File: octopus-common/src/main/java/octopus/teamcity/common/commonstep/CommonStepPropertyKeys.java (hunk 3 above)
Comment (rain-on) [reply octopus-common/src/main/java/octopus/teamcity/common/commonstep/CommonStepPropertyKeys.java]: resolved with "PropertyName" postfix

**Review by long-sean — APPROVED** [review]
Looks good to me

**Review by long-sean — APPROVED** [review]

**Pull request merged**